### ISSUE TITLE
Error - Expected 'Shutdown but found 'Booted' after waiting

### ISSUE THREAD

**Issue by nayakj** [calabash/calabash-ios#1270]
Hello Team,

I have recently upgraded the calabash-iso version to 0.20.4 (run_loop - 2.3.0).
and while running tests I am getting this error, the app is not launching at all  - 

*Expected 'Shutdown but found 'Booted' after waiting. (RuntimeError)
  ./features/support/01_launch.rb:33:in `start_app'
  ./features/support/01_launch.rb:55:in `Before'
onnection refused - Connection refused - connect(2) for "127.0.0.1" port 37265*

When I don't use RESET_BETWEEN_SCENARIOS=1 (along with the command to run the test), I do not get the above mentioned error. But I am using RESET_BETWEEN_SCENARIOS=1 since ages and the problem started only when I upgraded to 0.20.4.

**My start_app method has this**
launcher = Calabash::Cucumber::Launcher.new
launcher.relaunch
launcher.calabash_notify(self)

**When I run the test using binding.pry, i get the following error**
pry(#<Object>)> launcher = Calabash::Cucumber::Launcher.new
=> #<Launcher: not attached to an automator>
[2] pry(#<Object>)> launcher.relaunch
RuntimeError: Expected 'Shutdown but found 'Booted' after waiting.

#xcode installation location
$xcode-select --print-path
/Applications/Xcode.app/Contents/Developer

#xcode version
$Xcode 8.0
Build version 8A218a

#Calabash version
$calabash-ios version
0.20.4

#curl http://localhost:37265/version
{"device_family":"iPhone","outcome":"SUCCESS","server_port":37265,"simulator_device":"iPhone","simulator":"CoreSimulator 303.8 - Device: iPhone 6 - Runtime: iOS 9.3 (13E233) - DeviceType: iPhone 6","app_name":"My_App","app_version":"4.25.0.3207","screen_dimensions":{"native_scale":2,"sample":1,"height":1334,"width":750,"scale":2},"git":{"revision":"dc15597","remote_origin":"git@github.com:calabash\/calabash-ios-server.git","branch":"master"},"device_name":"iPhone Simulator","4inch":false,"app_id":"uk.co.bbc.iplayer","form_factor":"iphone 6","system":"x86_64","version":"0.20.4","iOS_version":"9.3","short_version_string":"4.25.0","ios_version":"9.3","iphone_app_emulated_on_ipad":false,"model_identifier":"iPhone7,2","app_base_sdk":"iphonesimulator10.1"}

Thanks in advance!

**Comment by QARajesh** [comment]
i am also seeing same error.

**Comment by RajeshReddy78** [comment]
I also seeing same error.

**Comment by duboviy** [comment]
Hi guys!

Thanks for your info, I've faced same problem too after upgrading to latest run_loop (2.3.0) and created following pull request https://github.com/calabash/run_loop/pull/600 to fix it.

Also I've created separate issue:
https://github.com/calabash/run_loop/issues/599
in [run_loop](https://github.com/calabash/) project to proceed work on it (as problem related to [run_loop](https://github.com/calabash/) repo)

If you don't want to wait for fix I may suggest you to use following workaround in your _Before_ block:

```ruby
Before do |scenario|
  launcher = Calabash::Launcher.launcher
  options = {
    :relaunch_simulator=>true,
    :reset_app_sandbox=>true
      # Add launch options here.
  }
  end
  launcher.relaunch(options)
end
```

**Comment by tim-burton** [comment]
Thanks for fix https://github.com/calabash/run_loop/pull/600
Also workaround is working fine for me!

**Comment by nayakj** [comment]
@duboviy 
I have added this in my before block now - 

 launcher = Calabash::Cucumber::Launcher.new
  options = {
      :relaunch_simulator=>true,
      :reset_app_sandbox=>true
  }
  launcher.relaunch(options)
  launcher.calabash_notify(self)

I still get the same error(expected 'Shutdown but found 'Booted' after waiting. (RuntimeError)) for the first scenario(and only the first scenario). Following all scenarios run fine now
And everytime I execute/run feature file(with more than one scenarios) I get this error for the first scenario but rest all sceanrio run as expected.

Am I missing something?

**Comment by duboviy** [comment]
After you added options -> try to remove RESET_BETWEEN_SCENARIOS=1 (or set RESET_BETWEEN_SCENARIOS=0)

**Comment by nayakj** [comment]
That works!
Thank you so much @duboviy

**Comment by duboviy** [comment]
You are welcome!
I'm glad that it works for you! ;)

**Comment by jmoody** [comment]
Fixed in run-loop 2.3.1
